compiler/rustc_hir/src/definitions.rs

```diff
@@ -202,29 +228,7 @@ impl DefPath {
         let mut s = String::with_capacity(self.data.len() * 16);
 
         for component in &self.data {
-            write!(s, "::{}[{}]", component.data.as_symbol(), component.disambiguator).unwrap();
-        }
-
-        s
-    }
-
-    /// Returns a filename-friendly string for the `DefPath`, with the
-    /// crate-prefix.
-    pub fn to_string_friendly<F>(&self, crate_imported_name: F) -> String
-    where
-        F: FnOnce(CrateNum) -> Symbol,
-    {
-        let crate_name_str = crate_imported_name(self.krate).as_str();
-        let mut s = String::with_capacity(crate_name_str.len() + self.data.len() * 16);
-
-        write!(s, "::{}", crate_name_str).unwrap();
-
-        for component in &self.data {
-            if component.disambiguator == 0 {
-                write!(s, "::{}", component.data.as_symbol()).unwrap();
-            } else {
-                write!(s, "{}[{}]", component.data.as_symbol(), component.disambiguator).unwrap();
-            }
+            write!(s, "::{}", component).unwrap();
```

Comment:
`to_string_no_crate` is used in the `fmt::Debug` implementation of `DefId`, so I think this should use the verbose mode.

 It's not used in much else either, so you can add `_verbose` to its name.

Reply:
The implementation of `fmt::Display` for `DisambiguatedDefPathData` is already verbose but i'll rename the method to add `_verbose` suffix